src/components/PostDetails/PostDetails.jsx

```diff
@@ -1,45 +1,86 @@
-import React from 'react';
-import { NewCommentForm } from '../NewCommentForm';
+import React, { useState, useEffect } from 'react';
 import './PostDetails.scss';
+import PropTypes from 'prop-types';
+
+import { NewCommentForm } from '../NewCommentForm';
+import { getPostComments, deleteComment } from '../../api/comments';
+
+export const PostDetails = ({ postDetails, postId }) => {
+  const { title, body, id } = postDetails;
+  const [comments, setComments] = useState([]);
+  const [hideComent, setHideComment] = useState(true);
+
+  useEffect(() => {
+    getPostComments(postId)
+      .then(comments => setComments(comments));
+  }, [comments]);
+
+  function habdleButton() {
+    setHideComment(() => !hideComent);
+  }
+
+  async function deleteComments(commentId) {
+    await deleteComment(commentId);
+  }
+
+  return (
+    <div className="PostDetails">
+      <h2>Post details:</h2>
+      { id && (
+        <>
+          <section className="PostDetails__post">
+            <h3>{title}</h3>
+            <p>{body}</p>
+          </section>
+
+          <section className="PostDetails__comments">
+            <button
+              type="button"
+              className="button"
+              onClick={() => habdleButton()}
+            >
+              {
+                `${!hideComent ? `Show` : `Hide`} ${comments.length} commnets`
+              }
+            </button>
+
+            <ul className="PostDetails__list">
+              {
+                hideComent && comments.map(comment  => (
```

Comment:
hideComment is a name for a function

why do you show something if hideCommnet is true?